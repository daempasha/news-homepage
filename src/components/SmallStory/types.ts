import { Story } from "@types";

export interface SmallStoryProps extends Story {
    index: number;
}