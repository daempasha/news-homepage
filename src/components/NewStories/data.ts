export default [
    {
        title: "Hydrogen Vs Electric Cars",
        description: "Will hydrogen-fueled cars ever catch up to EVs?"
    },
    {
        title: "The Downsides of AI Artistry",
        description: "What are the possible adverse affects of on-demand AI image generation?"
    },
    {
        title: "Is VC Funding Drying Up",
        description: "Private funding by VC firms is down 50% YOY. We take a look at what that means."
    }
]