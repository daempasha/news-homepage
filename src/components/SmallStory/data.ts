import revivingRetroPcImage from "@assets/image-retro-pcs.jpg";
import imageGamingGrowthImage from "@assets/image-gaming-growth.jpg";
import topLaptopsImage from "@assets/image-top-laptops.jpg";

export default [
    {
        image: revivingRetroPcImage,
        title: "Reviving Retro PCs",
        description: "What happens when old PCs are given modern upgrades?"
    },
    {
        image: topLaptopsImage,
        title: "Top 10 Laptops of 2022",
        description: "Our best picks for various needs and budgets."
    },
    {
        image: imageGamingGrowthImage,
        title: "The Growth of Gaming",
        description: "How the pandemic has sparked fresh opportunities."
    }
]