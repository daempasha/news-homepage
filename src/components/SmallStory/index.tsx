import { SmallStoryProps } from "./types";

const SmallStory: React.FC<SmallStoryProps> = ({ index, title, description, image }): React.ReactElement => {
    return (
        <div className="grid items-center grid-flow-dense grid-cols-12 gap-5 my-5 mx-2">
            <div className="image col-span-5 md:col-span-5  lg:col-span-3  ">
                <img className="h-32" src={image} alt={title} />
            </div>
            <div className="col-span-7 md:col-span-7 lg:col-span-9">
                <h1 className="text-3xl text-gray-300 font-bold ">{`${(index + 1)}`.padStart(2, "0")}</h1>
                <a className=" font-bold text-lg lg:text-xl my-2 cursor-pointer hover:text-orange-500">{title}</a>
                <p className="text-sm lg:text-base text-gray-600">{description}</p>
            </div>
        </div>
    );
}

export default SmallStory;