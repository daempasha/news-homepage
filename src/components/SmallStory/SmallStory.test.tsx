import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import data from './data';
import SmallStory from '.';





describe('<NewStories />', () => {
    it('renders correctly', () => {
        const newStories = render(<SmallStory index={0} {...data[0]} />)
        expect(newStories).toMatchSnapshot()
    });

});