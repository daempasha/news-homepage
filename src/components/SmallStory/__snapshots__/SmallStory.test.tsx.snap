// Vitest Snapshot v1

exports[`<NewStories /> > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid items-center grid-flow-dense grid-cols-12 gap-5 my-5 mx-2"
      >
        <div
          class="image col-span-5 md:col-span-5  lg:col-span-3  "
        >
          <img
            alt="Reviving Retro PCs"
            class="h-32"
            src="/src/assets/image-retro-pcs.jpg"
          />
        </div>
        <div
          class="col-span-7 md:col-span-7 lg:col-span-9"
        >
          <h1
            class="text-3xl text-gray-300 font-bold "
          >
            01
          </h1>
          <a
            class=" font-bold text-lg lg:text-xl my-2 cursor-pointer hover:text-orange-500"
          >
            Reviving Retro PCs
          </a>
          <p
            class="text-sm lg:text-base text-gray-600"
          >
            What happens when old PCs are given modern upgrades?
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid items-center grid-flow-dense grid-cols-12 gap-5 my-5 mx-2"
    >
      <div
        class="image col-span-5 md:col-span-5  lg:col-span-3  "
      >
        <img
          alt="Reviving Retro PCs"
          class="h-32"
          src="/src/assets/image-retro-pcs.jpg"
        />
      </div>
      <div
        class="col-span-7 md:col-span-7 lg:col-span-9"
      >
        <h1
          class="text-3xl text-gray-300 font-bold "
        >
          01
        </h1>
        <a
          class=" font-bold text-lg lg:text-xl my-2 cursor-pointer hover:text-orange-500"
        >
          Reviving Retro PCs
        </a>
        <p
          class="text-sm lg:text-base text-gray-600"
        >
          What happens when old PCs are given modern upgrades?
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
